ribs/emitters/opt/_evolution_strategy_base.py

```diff
@@ -1,5 +1,87 @@
 """Provides EvolutionStrategyBase."""
+from abc import ABC, abstractmethod
 
+import numpy as np
 
-class EvolutionStrategyBase:
-    pass
+
+class EvolutionStrategyBase(ABC):
+    """Base class for evolution strategy optimizers for use with emitters.
+
+    The basic usage is:
+    - Initialize the optimizer and reset it.
+    - Repeatedly:
+      - Request new solutions with ``ask()``
+      - Rank the solutions in the emitter (better solutions come first) and
+        pass them back with ``tell()``.
+      - Use ``check_stop()`` to see if the optimizer has reached a stopping
+        condition, and if so, call ``reset()``.
+
+    Args:
+        sigma0 (float): Initial step size.
+        solution_dim (int): Size of the solution space.
+        batch_size (int): Number of solutions to evaluate at a time. If None, we
+            calculate a default batch size based on solution_dim.
+        seed (int): Seed for the random number generator.
+        dtype (str or data-type): Data type of solutions.
+    """
+
+    def __init__(self,
+                 sigma0,
+                 solution_dim,
+                 batch_size=None,
+                 seed=None,
+                 dtype=np.float64):
+        self.batch_size = (4 + int(3 * np.log(solution_dim))
+                           if batch_size is None else batch_size)
+        self.sigma0 = sigma0
```

Comment:
Similar to `_gradient_opt_base`, I think we should do nothing in this constructor and leave it up to users to handle these arguments.

Reply:
For instance, initializing `self._rng` here is confusing because if I go to CMA-ES and look at `self._rng` being used in `ask()`, I have no clue where it was initialized. If we want to make a better supported API for ES and gradient optimizers, we should look at how we did it for `EmitterBase` and `ArchiveBase`, but I don't think it's worth it to go to that trouble for a supporting class like this.